src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Listeners/EventHubListener.cs

```diff
@@ -1,330 +1,390 @@
+                    {
+                        _cts.Dispose();
+                    }
+
+                    _disposed = true;
+                }
+            }
+
+            public void Dispose()
+            {
+                Dispose(true);
+            }
+
+            async Task ICheckpointer.CheckpointAsync(PartitionContext context)
+            {
+                await context.CheckpointAsync();
+            }
+
+            private Dictionary<string, object> GetLinksScope(EventData message)
+            {
+                if (TryGetLinkedActivity(message, out var link))
+                {
+                    return new Dictionary<string, object> {["Links"] = new[] {link}};
+                }
+
+                return EmptyScope;
+            }
+
+            private Dictionary<string, object> GetLinksScope(EventData[] messages)
+            {
+                List<Activity> links = null;
+
+                foreach (var message in messages)
+                {
+                    if (TryGetLinkedActivity(message, out var link))
+                    {
+                        if (links == null)
+                        {
+                            links = new List<Activity>(messages.Length);
+                        }
+
+                        links.Add(link);
+                    }
+                }
+
+                if (links != null)
+                {
+                    return new Dictionary<string, object> {["Links"] = links};
+                }
+
+                return EmptyScope;
+            }
+
+            private bool TryGetLinkedActivity(EventData message, out Activity link)
+            {
+                link = null;
+
+                if (((message.SystemProperties != null && message.SystemProperties.TryGetValue("Diagnostic-Id", out var diagnosticIdObj)) || message.Properties.TryGetValue("Diagnostic-Id", out diagnosticIdObj)) 
+                    && diagnosticIdObj is string diagnosticIdString)
+                {
+                    link = new Activity("Microsoft.Azure.EventHubs.Process");
+                    link.SetParentId(diagnosticIdString);
+                    return true;
+                }
+
+                return false;
+            }
+
+            private string GetOperationDetails(PartitionContext context, string operation, IDictionary<string, string> triggerDetails = null)
+            {
+                JObject contextJO = new JObject()
+                {
+                    { "partitionId", context.PartitionId },
+                    { "owner", context.Owner },
+                    { "eventHubPath", context.EventHubPath }
+                };
+
+                JObject rootJO = new JObject()
+                {
+                    { "operation", operation },
+                    { "partitionContext", contextJO }
+                };
+
+                if (triggerDetails != null)
+                {
+                    rootJO["triggerDetails"] = JObject.FromObject(triggerDetails);

```

Comment:
Do we need to log the trigger details again? They should be auto-logged by setting them in `TriggeredFunctionData`